### ISSUE TITLE
Only debounce Google URLs when SafeBrowsing is disabled

### ISSUE THREAD

**Issue by ryanbr** [brave/brave-browser#20084]
## Description 
Debounce embedded google urls without targeting safebrowsing urls.  

## Steps to Reproduce

   1. Received a tracking ticket on gmail, with the following google bounce
   2. `https://www.google.com/url?q=https://t.17track.net/%23nums%3D2222062731&source=gmail&ust=2239542218716022&usg=A12Vaw22aVcgCnimxpL3T22Gs--w`
   3.  Able to target this bounce without causing safebrowsing urls.

## Miscellaneous Information:
Original PR: https://github.com/brave/adblock-lists/pull/728 (And reverted https://github.com/brave/adblock-lists/pull/729)

**Comment by pes10k** [comment]
Thanks very much @ryanbr !

@pilgrim-brave , previously we discussed only debouncing Google URLs like these when SafeBrowsing is disabled. Let's use this issue to track that project.  I'll assign now, but let's discuss whenever is convenient where in the priority queue this slots.

**Comment by fmarier** [comment]
Noting that this was discussed during the privacy meeting on 2022-06-07 and we were not able to find a way to debounce the URLs coming from Apple Mail without affecting outgoing links from GMail (web UI) or Google Docs.

The security team's opinion is that disabling this security-relevant bouncer for people without Safe Browsing is not a good tradeoff since users without Safe Browsing are even more vulnerable to phishing/malware links.

Perhaps we should close this issue until we can think of another way to address this?

**Comment by pes10k** [comment]
after thinking this through again, i do not think we came to the right conclusion. I don't think we should keep applying (effectively) Google's safebrowsing to users who have opted out of Google's safe browsing. I appreciate that the security team thinks that its unwise for users to disable safebrowsing, but if thats the case we should remove the ability for users to opt out of safebrowsing (note, i do not think we should do this).

But, continuing to apply safebrowsing (i.e. the google redirect) sometimes because we think the user shouldn't have disabled it at all seems both confusing and unkind to our users.

I appreciate others don't agree, but please leave this open until we can discuss at the next privacy confab then

**Comment by thypon** [comment]
https://user-images.githubusercontent.com/581115/177705756-253d279c-4aed-430e-badf-6c41d0d1118a.mp4

The video shows that the `ping` URL is never hit to redirect the user when they click on the link.

**Comment by pes10k** [comment]
This is not correct. As discussed in slack and in privacy confab this bouncer is used to modify links in gmail accounts when those gmail accounts are accessed through 3p software (mail.app, thunderbird, etc)

**Comment by bsclifton** [comment]
@pes10k @ryanbr is this still relevant?

**Comment by ryanbr** [comment]
@fmarier might have some thoughts here

**Comment by pes10k** [comment]
I still think it would be a good change (to have a way to avoid Google servers in these user configurations), but I think most folks felt otherwise, so I think it’s prob safe to close

**Comment by fmarier** [comment]
We'll have the ability for users to add custom debounce rules in https://github.com/brave/brave-browser/issues/22996 which should give users more agency over what rules they apply in their browser, beyond the ones we apply by default.
